### PR TITLE
update macro used for passing through all columns to ensure quoting

### PR DESCRIPTION
**Pull request by fivetran-reneeli**
## PR Overview
**This PR will address the following Issue/Feature:**
https://github.com/fivetran/dbt_hubspot_source/issues/128
**This PR will result in the following new package version:** 0.16.0

While changes are made 'behind the scenes' to now allow models to successfully run with both `hubspot__pass_through_all_columns` and `hubspot__<>_pass_through_columns`, this may be a breaking change due to leveraging the `remove_duplicate_and_prefix_from_columns` macro. This is a breaking change because this macro can remove duplicate fields, resulting in an impact to your schema. See the [v0.12.0 release notes](https://github.com/fivetran/dbt_hubspot_source/releases/tag/v0.12.0) for more information.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Breaking Changes
- Switched from using the `fivetran_utils.remove_prefix_from_columns` macro to the `hubspot_source.remove_duplicate_and_prefix_from_columns` macro for when `hubspot__pass_through_all_columns` is enabled and you are passing through all columns in the `stg_hubspot__company`, `stg_hubspot__contact`, `stg_hubspot__deal`, and `stg_hubspot__ticket` models. This also ensures the source fields passed through are all quoted from the onset. This is a breaking change because this macro can remove duplicate fields, resulting in an impact to your schema. See the [v0.12.0 release notes](https://github.com/fivetran/dbt_hubspot_source/releases/tag/v0.12.0) for more information.

## Bug Fixes
- Introduced hubspot-specific version of the `fivetran_utils.pass_through_columns` macro titled `hubspot_add_pass_through_columns`, which introduces quoting around the source fields being brought in as passthrough columns. This will ensure that your warehouse reads the sql correctly, particularly if the field contains special characters or syntax.

## Under the Hood
- Updated seed data to include fields with special syntax in order to test the above changes.

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [ ] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

see internal ticket

### If you had to summarize this PR in an emoji, which would it be?
 
:dancer:

### CODE REVIEW

File: macros/hubspot_add_pass_through_columns.sql
Comment (fivetran-joemarkiewicz) [inline macros/hubspot_add_pass_through_columns.sql:14]:
```suggestion
{% macro hubspot_add_pass_through_columns(base_columns, pass_through_var) %}

  {% if pass_through_var %}

    {% for column in pass_through_var %}

    {% if column is mapping %}

      {% if column.alias %}

        {% do base_columns.append({ "name": column.name, "alias": column.alias, "quote": True, "datatype": column.datatype if column.datatype else dbt.type_string()}) %}

      {% else %}

        {% do base_columns.append({ "name": column.name, "quote": True, "datatype": column.datatype if column.datatype else dbt.type_string()}) %}
        
      {% endif %}

    {% else %}

      {% do base_columns.append({ "name": column, "quote": True, "datatype": dbt.type_string()}) %}

    {% endif %}

    {% endfor %}

  {% endif %}

{% endmacro %}
```

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:3]: Suggestion to just make this more generalized.
```suggestion
## Bug Fixes
```

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:5]:
```suggestion
- Switched from using the `fivetran_utils.remove_prefix_from_columns` macro to the `hubspot_source.remove_duplicate_and_prefix_from_columns` macro for when `hubspot__pass_through_all_columns` is enabled and you are passing through all columns in the `stg_hubspot__company`, `stg_hubspot__contact`, `stg_hubspot__deal`, and `stg_hubspot__ticket` models. Similarly, this ensures the source fields passed through are all quoted.
```

File: CHANGELOG.md
Comment (fivetran-reneeli) [reply CHANGELOG.md:12]: updated

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:5]: I don't think we necessarily need to link the v0.12.0 release notes since that may cause more confusion. I think what you have before is great, no need to add the last sentence.

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:8]:
```suggestion
- Introduced hubspot-specific version of the `fivetran_utils.add_pass_through_columns` macro titled `hubspot_add_pass_through_columns`, which introduces quoting around the source fields being brought in as passthrough columns. This will ensure that your warehouse reads the sql correctly, particularly if the field contains special characters or syntax.
```

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [reply CHANGELOG.md:5]: Also the `stg_hubspot__ticket` model was not updated in this release for this change. You can include this in the below bugfix however.

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [reply CHANGELOG.md:8]: Also, please list out the impacted models.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-reneeli LGTM with a few final CHANGELOG change requests. Nothing required to block the approval and move to release review. Thanks for working through these updates!

File: CHANGELOG.md
Comment (fivetran-reneeli) [reply CHANGELOG.md]: done

**Comment by fivetran-reneeli** [comment]
Thanks @fivetran-joemarkiewicz made the changes to the CHANGELOG. Will move on to release review

File: integration_tests/dbt_project.yml
Comment (fivetran-avinash) [inline integration_tests/dbt_project.yml:15]: Should lines 15-16 be commented out before merging?

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-reneeli Nice work, particularly with the cross-warehouse seed testing. Just a few quick comments before approval!

File: integration_tests/dbt_project.yml
Comment (fivetran-reneeli) [reply integration_tests/dbt_project.yml:71]: No particular reason for keeping them in-- will comment it out

**Review by fivetran-avinash — APPROVED** [review]
@fivetran-reneeli lgtm!

**Pull request merged**
